### PR TITLE
parsec-bin: update runtime dependencies

### PR DESCRIPTION
**Pull request by arcnmx**
## Description of changes

parsecd-150-89b now requires curl, png, and jpeg8

Note that there is no change to the initial install package (`parsec-bin.src`) - these are only required by the dylib it downloads as part of its auto-update process, but I've updated the URL with the latest public download link hostname.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NobbZ** [comment]
In the unoficial Discord we today realized that there is also a shared object in `~/.parsec` that might refer to invalid store paths over time.

A user had the problem that parsecd didn't work for their user who used it before, but it started as a different user, which did not have had that shared object before.

Do you think you can tackle that issue somehow in this PR, by blindly copying over shared objects from the skeleton folders or avoiding the necessity to copy by providing the skeleton in the `RPATH`/`LD_LIBRARY_PATH`?

I do not use this software myself, though found this issue because we had that topic in the chat just a couple of minutes ago.

This is not a formal review, but a regular question, but I'd by happy to get you through the review process either way after you quickly replied here.

**Comment by BobTheBob9** [comment]
yeah, at the moment it seems necessary to remove and regenerate ~/.parsec if binaries change at all, would be nice if this wasn't required

**Comment by arcnmx** [comment]
> yeah, at the moment it seems necessary to remove and regenerate ~/.parsec if binaries change at all, would be nice if this wasn't required

It [does use LD_LIBRARY_PATH](https://github.com/NixOS/nixpkgs/blob/6cbba87da11bd5c5fec58941cee96959469b7926/pkgs/applications/misc/parsec/bin.nix#L66-L68) and does not patch the files downloaded into `~/.parsec`, so that generally shouldn't happen (`strings ~/.parsec/* | grep -F /nix` should indeed only produce log entries if anything at all)

It does seem possible that the original/old dylib from the `.deb` that gets copied from `share/parsec/skel` in the nix store is patched though (it really shouldn't be), then if a user lacks internet access to update it to an unpatched dylib (does `parsecd` not perform the update before loading the dylib? I assumed it checks for updates before launching), then performs a new switch+gc, it could feasibly get stuck? But *if* it does update first, that should be a mostly-impossible scenario - did the user check if the permissions on the folder were broken maybe and preventing it from updating? Or maybe a firewall was blocking the update?

Either way, `share/parsec/skel` should be in an autopatchelf blacklist or something, only the main binary needs patching as long as it's run with the wrapper script.

File: pkgs/applications/misc/parsec/bin.nix [hunk 1]
```diff
@@ -4,18 +4,18 @@
 , fetchurl
 , alsa-lib, openssl, udev
 , libglvnd
-, libX11, libXcursor, libXi, libXrandr
+, libX11, libXcursor, libXi, libXrandr, libXfixes
 , libpulseaudio
 , libva
-, ffmpeg
+, ffmpeg, libpng, libjpeg8, curl
```
Comment (Atemu) [inline pkgs/applications/misc/parsec/bin.nix:10]: Nit: Could you put these each on its own line?

File: pkgs/applications/misc/parsec/bin.nix [hunk 2]
```diff
@@ -74,6 +78,16 @@ stdenvNoCC.mkDerivation {
...
+  fixupPhase = ''
+    runHook preFixup
+
+    autoPatchelf $out/bin
+
+    runHook postFixup
+  '';
+
+  dontAutoPatchelf = true;
+
```
Comment (Atemu) [inline pkgs/applications/misc/parsec/bin.nix:90]: Please add a comment why this is necessary. Commit messages are nice but it's better to have this sort of stuff in-code.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -34393,6 +34393,7 @@ with pkgs;
 
   parsec-bin = callPackage ../applications/misc/parsec/bin.nix {
     ffmpeg = ffmpeg_4;
+    libjpeg8 = libjpeg.override { enableJpeg8 = true; };
```
Comment (Atemu) [inline pkgs/top-level/all-packages.nix:34396]: This is used >2 times, we should make this a top-level attribute.

**Review by pabloaul — APPROVED** [review]
tested and works connecting to my machine :+1:

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -30488,10 +30486,10 @@ with pkgs;
   };
   bitwig-studio3 =  callPackage ../applications/audio/bitwig-studio/bitwig-studio3.nix { };
   bitwig-studio4 =  callPackage ../applications/audio/bitwig-studio/bitwig-studio4.nix {
-    libjpeg = libjpeg.override { enableJpeg8 = true; };
+    libjpeg = libjpeg8;
   };
   bitwig-studio5 =  callPackage ../applications/audio/bitwig-studio/bitwig-studio5.nix {
-    libjpeg = libjpeg.override { enableJpeg8 = true; };
+    libjpeg = libjpeg8;
```
Comment (Atemu) [inline pkgs/top-level/all-packages.nix:30492]: Nit: I usually prefer packages to depend on the version inside the package definition's function args.

**Review by Atemu — APPROVED** [review]

**Pull request merged**